### PR TITLE
Fix artifacts for build:vio by merging build:vio and validate:vio

### PR DESCRIPTION
**Pull request by SkySkimmer**
https://github.com/coq/coq/pull/17785 changed the artifact paths but
this job wasn't adapted.

Because of the workaround for
https://gitlab.com/gitlab-org/gitlab/issues/202505 this wasn't detected.

In this PR we also try removing the workaround, maybe the problem is fixed?

### CODE REVIEW

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by SkySkimmer** [comment]
It faled in light CI https://gitlab.inria.fr/coq/coq/-/jobs/3673643 so there's no point running full ci

**Comment by SkySkimmer** [comment]
@Zimmi48 why are we using the :dev template for build:vio?

**Comment by Zimmi48** [comment]
No idea. This is the case since #15560, so we should ask @ejgallego. Cf. https://github.com/coq/coq/pull/15560/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R292.

**Comment by SkySkimmer** [comment]
I guess it's because regular build template use `dune build -p` so would be harder to make it use vio.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -332,7 +332,6 @@ build:base+async:
 build:vio:
   extends: .build-template:base:dev
   variables:
-    COQ_EXTRA_CONF: "-native-compiler no"
```
Comment (ejgallego) [inline .gitlab-ci.yml:335]: Why this change?

Comment (SkySkimmer) [reply .gitlab-ci.yml:335]: COQ_EXTRA_CONF is for build-template not build-template:base:dev AFAICT

**Comment by SkySkimmer** [comment]
>In this PR we also try removing the workaround, maybe the problem is fixed?

It's not fixed, cf https://gitlab.inria.fr/coq/coq/-/jobs/3688371

**Comment by SkySkimmer** [comment]
Maybe we should just merge the jobs

**Comment by ejgallego** [comment]
> Maybe we should just merge the jobs

We can just remove `.vio` at all, as we discussed in the call. Given that vio files are quadratic in size, I pretty much dunno they are useful at all.

**Comment by SkySkimmer** [comment]
>We can just remove .vio at all, as we discussed in the call.

That doesn't seem like it will happen short term, maybe we should still merge this PR.

**Comment by ejgallego** [comment]
I don't understand this issue, letting other take care of it.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by ejgallego** [comment]
Note that #18424 is also ready for merge.
